sklearn/metrics/_plot/tests/test_confusion_matrix_display.py

```diff
@@ -0,0 +1,328 @@
+    X, y = data
+    regressor = SVR().fit(X, y)
+
+    err_msg = "ConfusionMatrixDisplay.from_estimator only supports classifiers"
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_estimator(regressor, X, y)
+
+    y_pred_regressor = regressor.predict(X)
+
+    err_msg = (
+        "Only 'binary' and 'multiclass' classification problems are supported"
+    )
+    with pytest.raises(ValueError, match=err_msg):
+        # Force `y_true` to be seen as a regression problem
+        ConfusionMatrixDisplay.from_predictions(y + 0.5, y_pred)
+
+    err_msg = "Got `y_pred` of type 'continuous'"
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_predictions(y, y_pred_regressor)
+
+    err_msg = "Found input variables with inconsistent numbers of samples"
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_predictions(y, y_pred[::2])
+
+
+def test_confusion_matrix_display_invalid_option(
+    pyplot, confusion_matrix_display_fxt
+):
+    constructor, *params = confusion_matrix_display_fxt
+    extra_params = {"normalize": "invalid"}
+
+    err_msg = r"normalize must be one of \{'true', 'pred', 'all', None\}"
+    with pytest.raises(ValueError, match=err_msg):
+        constructor(*params, **extra_params)
+
+
+@pytest.mark.parametrize("with_labels", [True, False])
+@pytest.mark.parametrize("with_display_labels", [True, False])
+def test_confusion_matrix_display_custom_labels(
+    pyplot,
+    confusion_matrix_display_fxt,
+    data,
+    y_pred,
+    n_classes,
+    with_labels,
+    with_display_labels,
+):
+
+    X, y = data
+    constructor, *params = confusion_matrix_display_fxt
+
+    ax = pyplot.gca()
+    labels = [2, 1, 0, 3, 4] if with_labels else None
+    display_labels = ["b", "d", "a", "e", "f"] if with_display_labels else None
+
+    cm = confusion_matrix(y, y_pred, labels=labels)
+    disp = constructor(
+        *params,
+        ax=ax,
+        display_labels=display_labels,
+        labels=labels
+    )
+    assert_allclose(disp.confusion_matrix, cm)
+
+    if with_display_labels:
+        expected_display_labels = display_labels
+    elif with_labels:
+        expected_display_labels = labels
+    else:
+        expected_display_labels = list(range(n_classes))
+
+    expected_display_labels_str = [str(name)
+                                   for name in expected_display_labels]
+
+    x_ticks = [tick.get_text() for tick in disp.ax_.get_xticklabels()]
+    y_ticks = [tick.get_text() for tick in disp.ax_.get_yticklabels()]
+
+    assert_array_equal(disp.display_labels, expected_display_labels)
+    assert_array_equal(x_ticks, expected_display_labels_str)
+    assert_array_equal(y_ticks, expected_display_labels_str)
+
+
+@pytest.mark.parametrize("normalize", ["true", "pred", "all", None])
+@pytest.mark.parametrize("include_values", [True, False])
+def test_confusion_matrix_display_plotting(
```

Comment:
I know it was like that before, but I think that short comments describing what the check is doing would really help reviewing, and also our future selves. This is typically the second time I'm reviewing this check and I have to read it all again to understand what it's doing. I wish we could enforce this somehow. CC @thomasjpfan ;)

Reply:
If we move to docstrings instead of comments to document our tests, we can enforce this. i.e.:

```py
def test_something_cool():
	"""Checks something cool"""
```

I have been wanting to do this for a while. It may also make for a good sprint issue.

Reply:
The issue with docstring was with `nose` (it was overwriting the name of the test). With `pytest` I don't think that there is anything stopping us to make a move toward this solution.